src/java/org/apache/cassandra/cache/AutoSavingCache.java

```diff
@@ -196,12 +205,18 @@ public int loadSaved()
         // modern format, allows both key and value (so key cache load can be purely sequential)
         File dataPath = getCacheDataPath(CURRENT_VERSION);
         File crcPath = getCacheCrcPath(CURRENT_VERSION);
-        if (dataPath.exists() && crcPath.exists())
+        File metadataPath = getCacheMetadataPath(CURRENT_VERSION);
+        if (dataPath.exists() && crcPath.exists() && metadataPath.exists())
```

Comment:
This is doable, but I don't know if complicating this code is worth restoring the cache from previous version 🤷🏻‍♂️

Reply:
Given that this change already landed, I don't mind this being punted from this PR, but would be good to ask in dev@?  This would be a performance issue during major upgrades but once warm you stop seeing it...